QWeb/internal/util.py

```diff
@@ -246,6 +257,13 @@ def _handle_old_style_prefs(prefs: str) -> dict:
     return d
 
 
+def validate_prefs(prefs: Optional[Any]) -> dict:
+    if isinstance(prefs, dict):
+        return prefs
+
+    return prefs_to_dict(str(prefs).strip())
+

```

Comment:
It took me a while to figure out what is the purpose of this method and I'm guessing it's only for typechecking.

It would probably be better to loosen up the type definitions of `prefs_to_dict()` and add the str casting into it.

The typechecker can't actually know what is the actual type of `prefs` as it comes from a dictionary even though it should always be either str or dict.

```
def prefs_to_dict(prefs: Any) -> dict[str, Any]:
    if isinstance(prefs, dict):
        d = prefs
    else:
        prefs_j = "{" + str(prefs).strip() + "}"
        try:
            d = json.loads(prefs_j)
        except json.decoder.JSONDecodeError:
            try:
                d = _handle_old_style_prefs(prefs)
            except QWebUnexpectedConditionError as e:
                raise QWebUnexpectedConditionError(
                    'Invalid argument! Experimental opts should given in robot dict '
                    'or string in format: key1:value1, key2:value2') from e
    return d
```

Reply:
not changed in this pr, so will make a specific refactor task about this

Reply:
sorry, yep the new function was added in this PR. There was a reason, lint complained about too many branches, so I needed to move parts to own util func.

Reply:
changed validate_prefs to parse_prefs